haproxy/datadog_checks/haproxy/haproxy.py

```diff
@@ -537,9 +537,14 @@ def _process_backend_hosts_metric(
                 agg_statuses[service]
 
         for service in agg_statuses:
-            tags = ['service:%s' % service]
+            tags = ['haproxy_service:%s' % service]
             tags.extend(custom_tags)
             tags.extend(active_tag)
+
+            if not self.instance.get('disable_legacy_service_tag', False):
+                self._log_deprecation('service_tag', 'haproxy_service')
+                tags.append('service:{}'.format(service))
```

Comment:
Can we extract that as util method in base to avoid duplication ?

```
            if not self.instance.get('disable_legacy_service_tag', False):
                self._log_deprecation('service_tag', 'haproxy_service')
                tags.append('service:{}'.format(service))
```

Reply:
Good idea, mind if I do that in a future PR with other service tag work?

Reply:
I think it worth refactoring, at lest having a function in this integration. There is 7 repetition of this block.